libc/getpw.c

```diff
@@ -52,8 +52,14 @@ TEST(getpwd, getpwnam_getroot)
 		TEST_ASSERT_EQUAL_INT(0, pw->pw_uid);
 		TEST_ASSERT_EQUAL_INT(0, pw->pw_gid);
 		TEST_ASSERT_EQUAL_STRING("root", pw->pw_gecos);
+		/* "/" is a root dir and the supported shell is sh on Phoenix-RTOS */
+#ifdef __phoenix__
 		TEST_ASSERT_EQUAL_STRING("/", pw->pw_dir);
 		TEST_ASSERT_EQUAL_STRING("/bin/sh", pw->pw_shell);
+#else
+		TEST_ASSERT_EQUAL_STRING("/root", pw->pw_dir);
+		TEST_ASSERT_EQUAL_STRING("/bin/bash", pw->pw_shell);
+#endif
```

Comment:
How about using two macros for these  strings which are selected based on proposed `ifdef`? Less `ifdef`s in the test body